Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/RayTracing/RayTracingIntersectionAttributes.azsli

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+// This structure must not be larger than 32 bytes (D3D12_RAYTRACING_MAX_ATTRIBUTE_SIZE_IN_BYTES)
+// TODO(intersection): Which attributes are useful here? Need to pack attributes?
```

Comment:
Not fixes, what would be good attributes for this? Since positon+normal+uv are already 32 bytes, adding tangent+bitangent would require compression of some sort.
Using half-float for everything would allow everything to fit (pos+normal+uv+tangent+bitangent = (3+3+2+4+3)*2 = 30 bytes. Is half-float sufficient for this?

Reply:
The data in `ProceduralGeometryIntersectionAttributes` is only passed from intersection shaders to the hit shader, so for this purpose it is possible to use half float. Since it is also the easiest option to be able to store all 5 attributes, I changed the types to half and downcast them in the intersection shaders and upcast them again in the closesthit shader. I hope this is satisfactory for now.